app/main.py

```diff
@@ -14,14 +28,18 @@ def main():
     q = request.args.get('q')
     if not q:
         abort(400)
+    key = '_'.join([
+        conf.KEY_PREFIX,
+        hashlib.md5(str(q).encode('utf-8')).hexdigest()
```

Comment:
Hmm, this doesn't seem quite right. Unicode input doesn't seem to be handled cleanly:

<img width="744" alt="screenshot 2016-01-29 13 00 44" src="https://cloud.githubusercontent.com/assets/96396/12688028/557149be-c688-11e5-89ef-ddf51874b456.png">


Traceback:
```
nginx_1     | 192.168.99.1 - - [29/Jan/2016:21:00:11 +0000] "GET /?q=%F0%9F%92%94 HTTP/1.1" 202 2 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.10; rv:47.0) Gecko/20100101 Firefox/47.0" "-"
app_1       | [pid: 9|app: 0|req: 1/1] 172.17.0.5 () {42 vars in 679 bytes} [Fri Jan 29 21:00:11 2016] GET /?q=%F0%9F%92%94 => generated 2 bytes in 49 msecs (HTTP/1.0 202) 2 headers in 76 bytes (1 switches on core 0)
celery_1    | [2016-01-29 21:00:11,976: ERROR/MainProcess] Task main.recommend[ba4b1ddc-d01b-44b5-a331-52c9604f9dff] raised unexpected: IndexError('list index out of range',)
celery_1    | Traceback (most recent call last):
celery_1    |   File "/usr/local/lib/python3.5/site-packages/celery/app/trace.py", line 240, in trace_task
celery_1    |     R = retval = fun(*args, **kwargs)
celery_1    |   File "/app/util.py", line 12, in __call__
celery_1    |     return TaskBase.__call__(self, *args, **kwargs)
celery_1    |   File "/usr/local/lib/python3.5/site-packages/celery/app/trace.py", line 438, in __protected_call__
celery_1    |     return self.run(*args, **kwargs)
celery_1    |   File "/app/main.py", line 21, in recommend
celery_1    |     recommendation = SearchRecommendation(q).do_search(q)
celery_1    |   File "/app/search/recommendation.py", line 84, in do_search
celery_1    |     self.top_suggestion = self.get_top_suggestion(self.suggestions)
celery_1    |   File "/app/search/recommendation.py", line 55, in get_top_suggestion
celery_1    |     return suggestions[0]
celery_1    | IndexError: list index out of range
```